vtk_viewer/src/vtk_utils.cpp

```diff
@@ -394,6 +394,67 @@ vtkSmartPointer<vtkPolyData> estimateCurvature(vtkSmartPointer<vtkPolyData> mesh
   return curvature_filter->GetOutput();
 }
 
+void embedRightHandRuleNormals(vtkSmartPointer<vtkPolyData>& data)
+{
+  LOG4CXX_DEBUG(VTK_LOGGER, "Embedding mesh normals from right hand rule");
+  int size = data->GetNumberOfCells();
+  vtkDoubleArray* cell_normals = vtkDoubleArray::New();
+
+  cell_normals->SetNumberOfComponents(3);
+  cell_normals->SetNumberOfTuples(size);
+
+  // loop through all cells and add cell normals
+  for (int i = 0; i < size; ++i)
+  {
+    vtkCell* cell = data->GetCell(i);
+    if (cell)
+    {
+      // Get all point IDs associated with the given cell
+      vtkIdList* pts = cell->GetPointIds();
+      double norm[3] = { 0, 0, 0 };
+
+      // If there are at least 3 points associated with the cell
+      if (pts->GetNumberOfIds() >= 3)
+      {
+        // Define arrays to store the points
+        double p0[3] = { 1, 0, 0 };
+        double p1[3] = { 0, 1, 0 };
+        double p2[3] = { 0, 0, 1 };
+
+        // Extract points from polydata that are associated with this cell
+        data->GetPoint(pts->GetId(0), p0);
+        data->GetPoint(pts->GetId(1), p1);
+        data->GetPoint(pts->GetId(2), p2);
+
+        // Get vectors obeying RHR
+        double v1[3];
+        double v2[3];
+        for (int ind = 0; ind < 3; ind++)
+        {
+          v1[ind] = p1[ind] - p0[ind];
+          v2[ind] = p2[ind] - p1[ind];
+        }
+        // Normal is v1 x v2
+        norm[0] = v1[1] * v2[2] - v2[1] * v1[2];
+        norm[1] = -1 * (v1[0] * v2[2] - v2[0] * v1[2]);
+        norm[2] = v1[0] * v2[1] - v2[0] * v1[1];
```

Comment:
I think you may simplify a lot of this with the cross product function in Eigen

Reply:
At the time this seemed easier. They were already in arrays, I needed an array out, and the formula for a cross product isn't that long. If you feel strongly about it, I can convert it to Eigen and back again.

Reply:
The overhead of converting back and forth between eigen and c arrays may do away with the benefits of using eigen functions so let's leave it as it is. It'd be nice to do a comprehensive Eigen simplification at some point in the future.